src/workflows/multiomics/multisample/main.nf

```diff
@@ -1,155 +1,186 @@
-include { multisample_processing_workflow as multisample_processing_workflow } from workflowDir + '/multiomics/full_pipeline/main.nf'
-include { integration_setup_workflow as integration_setup_workflow } from workflowDir + '/multiomics/full_pipeline/main.nf'
-
-include { readConfig; helpMessage; readCsv; preprocessInputs; channelFromParams } from workflowDir + "/utils/WorkflowHelper.nf"
-include { setWorkflowArguments; getWorkflowArguments; passthroughMap as pmap; passthroughFlatMap as pFlatMap } from workflowDir + "/utils/DataflowHelper.nf"
-config = readConfig("$workflowDir/multiomics/multisample/config.vsh.yaml")
-
-workflow multisample {
-  helpMessage(config)
-
-  channelFromParams(params, config)
-    | run_wf
-
-}
+import java.util.concurrent.ConcurrentHashMap;
 
 workflow run_wf {
   take:
   input_ch
 
   main:
-  output_ch = input_ch
-    | preprocessInputs("config": config)
-    | setWorkflowArguments (
-        "split_modalities_args": [:],
-        "rna_multisample_args": [
-          "filter_with_hvg_var_output": "filter_with_hvg_var_output",
-          "filter_with_hvg_obs_batch_key": "filter_with_hvg_obs_batch_key",
-          "var_qc_metrics": "var_qc_metrics",
-          "top_n_vars": "top_n_vars"
-        ],
-        "prot_multisample_args": [:],
-          "integration_args_rna": [
-            "var_pca_feature_selection": "filter_with_hvg_var_output", // run PCA on highly variable genes only
-            "pca_overwrite": "pca_overwrite"
-          ],
-        "integration_args_prot": ["pca_overwrite": "pca_overwrite"],
-        "publish": ["output": "output"],
-    )
-    | getWorkflowArguments(key: "split_modalities_args")
-    | split_modalities_workflow
-    | pmap {id, data ->
-        def new_data = ["input": data.output]
-        [id, new_data]
-    }
-    | multisample_processing_workflow
-    | groupTuple(by: 0, sort: "hash")
-    | map { list -> 
-        def id = list[0]
-        def new_input = list[1].collect({it.output})
-        def other_arguments = list[2][0] // Get the first set, the other ones are copies
-        def modalities_list = ["modalities": list[-1].collect({it.modality}).unique()]
-        [id, ["input": new_input]] + other_arguments + modalities_list
+    multisample_ch = input_ch
+      // The input for this workflow can either be a list of unimodal files
+      // or a single multimodal file. To destingish between the two, the files will be split either way.
+      // For mutliple unimodal files, the result before or after splitting is identical.
+      // In both cases, this workflow requires split files.
+      
+      // Split must be called on each item of the input list, so split it into multiple events with unique ids
+      // Unique ids are required to run a component
+      | flatMap {id, state ->
+        def newEvents = state.input.withIndex().collect{input_file, index -> 
+          def newState = state + ["input": input_file]
```

Comment:
why not stash the original id in the state?

```suggestion
          def newState = state + ["input": input_file, "orig_id": id]

Reply:
Fixed in 7ddbf7f3a60fd6d5906eff610c332ae6fefeae78